SYCL/InvokeSimd/Regression/slm_load_store.cpp

```diff
@@ -0,0 +1,122 @@
+// TODO: enable on Windows once driver is ready
+// REQUIRES: gpu && linux
+// UNSUPPORTED: cuda || hip
+//
+// TODO: enable when Jira issue resolved
+// XFAIL: gpu
+//
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+//
+// VISALTO enable run
+// RUN: env IGC_VISALTO=63 IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+/*
+ * Test check basic support of local memory access in invoke_simd.
+ */
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/oneapi/experimental/invoke_simd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <functional>
+#include <iostream>
+#include <type_traits>
+
+/* Subgroup size attribute is optional
+ * In case it is absent compiler decides what subgroup size to use
+ */
+#ifdef IMPL_SUBGROUP
+#define SUBGROUP_ATTR
+#else
+#define SUBGROUP_ATTR [[intel::reqd_sub_group_size(VL)]]
+#endif
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+namespace esimd = sycl::ext::intel::esimd;
+constexpr int VL = 16;
+
+class Test {};
+using dtype = int;
+const int elem_num = 16;
+const int slm_size = 32 * 1024;
+
+__attribute__((always_inline)) void
+slm_load_store_test(nd_item<1> *ndi, dtype *A, dtype *B,
+                    dtype *C) SYCL_ESIMD_FUNCTION {
+  sycl::ext::intel::esimd::slm_init(slm_size);
+  esimd::simd<dtype, elem_num> src_vec;
+  src_vec.copy_from(A);
+  esimd::slm_block_store<dtype, elem_num>(0, src_vec);
+  esimd::simd<dtype, elem_num> dest_vec;
+  dest_vec.copy_from(C);
+  dest_vec = esimd::slm_block_load<dtype, elem_num>(0);
+  dest_vec.copy_to(C);
+}
+[[intel::device_indirectly_callable]] SYCL_EXTERNAL void __regcall invoke_slm_load_store_test(
+    nd_item<1> *ndi, dtype *A, dtype *B, dtype *C) SYCL_ESIMD_FUNCTION;
+
+int main(void) {
+  auto q = queue{gpu_selector_v};
+  auto dev = q.get_device();
+  auto ctxt = q.get_context();
+
+  std::cout << "Running on " << dev.get_info<sycl::info::device::name>()
+            << "\n";
+  std::cout << "Local Memory Size: "
+            << q.get_device().get_info<sycl::info::device::local_mem_size>()
+            << std::endl;
+  auto *A =
+      static_cast<dtype *>(malloc_shared(elem_num * sizeof(dtype), dev, ctxt));
+  auto *B =
+      static_cast<dtype *>(malloc_shared(elem_num * sizeof(dtype), dev, ctxt));
+  auto *C =
+      static_cast<dtype *>(malloc_shared(elem_num * sizeof(dtype), dev, ctxt));
```

Comment:
non-blocking comment: These lines can be shorter and more clear. If do so, you can also delete the variable 'ctxt'.
```suggestion
  auto *A = malloc_shared<dtype>(elem_num, q);
  auto *B = malloc_shared<dtype>(elem_num, q);
  auto *C = malloc_shared<dtype>(elem_num, q);
```